doc/lua_api.txt

```diff
@@ -3303,33 +3299,91 @@ The following functions provide escape sequences:
+and has a class to represent them (`vector.*`), which this chapter is about.
+For details on what a spatial vectors is, please refer to Wikipedia:
+https://en.wikipedia.org/wiki/Euclidean_vector.
+
+Spatial vectors are used for various things, including, but not limited to:
+
+* positions
+* position differences
+* velocities
+* accelerations
+* euler angles (Yes, using spatial vectors here has no real semantic meaning. It's
+  just 3 values. Therefore, most operations make no sense and should not be used
+  on vectors of euler angles.)
+* dimensions (Sizes in 3 directions.)
+
+Note that they are *not* used for:
+
+* n-dimensional vectors where n is not 3 (ie. n=2)
+* arrays of the form `{num, num, num}`
+
+The API documentation may refer to spatial vectors by any of the following notations:
+
+* `(x, y, z)` (Used rarely, and only if it's clear that it's a vector.)
+* `vector.new(x, y, z)`
+* `{x=num, y=num, z=num}`
+
+They are all (if not stated otherwise) referring to the same thing, that is: vectors,
+as produced by `vector.new`.
+
+Compatibility notes
+-------------------
+
+Internally, a vector is implemented using a table with the 3 fields `x`, `y` and `z`.
+It's similar to this: `{x = 0, y = 1, z = 0}`, but with a metatable.
+
+Historically (before Minetest 5.5.0), vectors used to be defined as tables of the
+form `{x = num, y = num, z = num}`, without any metatable. You might find this in
+old code.
+
+Most functions by Minetest that take vectors will continue to able to handle them,
+but passing such manually constructed tables to Minetest's or some other mod's
+functions (if they want a vector) is considered deprecated and is strongly
+discouraged.
+The rationale behind this is that Minetest's functions can be overridden by mods,
+or may pass given vectors as argument to some callback function. And the mod that
+gets the vector in the end may want to depend on having a vector with the here
+defined interface.
+Also, though not likely, the internal implementation of a vector might change in
 the future.
-Old code might still use vectors without metatables, be aware of this!
+In your own code, or if you define your own API, you can, of course, still use
+other representations of vectors.
+
+A mod should can expect to get vectors of this class whenever they come from the
+engine.
+In the future, the same should apply for vectors coming from other mods.
+
+Special properties of the class
+-------------------------------
+
+Vectors can be indexed with numbers and allow method and operator syntax.
 
 All these forms of addressing a vector `v` are valid:
 `v[1]`, `v[3]`, `v.x`, `v[1] = 42`, `v.y = 13`
+(Though using letters over numbers is preferable due to performance and compatibility
+reasons.)
 
 Where `v` is a vector and `foo` stands for any function name, `v:foo(...)` does
 the same as `vector.foo(v, ...)`, apart from deprecated functionality.
 
+Where `v` is a vector, `tostring(v)` does the same as `v:to_string()`.
```

Comment:
IMO this belongs to the `to_string()` documentation.

Reply:
It's now at both places, because it's also a special property of the class.